clippy_lints/src/utils/ty.rs

```diff
@@ -0,0 +1,16 @@
+use rustc::ty::Ty;
+
+fn is_internally_mutable(ty: Ty) {
```

Comment:
Oops, that was an early try to create a utility function that would help us find out if a function is pure (together with finding global state). I thought I'd stashed that. Egad, I'll remove it from the PR.